File: inject-java/src/test/groovy/io/micronaut/aop/itfce/InterfaceImpl.java
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.aop.itfce;

import javax.inject.Singleton;
import java.util.Collections;
import java.util.List;

/**
 * @author Graeme Rocher
 * @since 1.0
 */
@Singleton
public class InterfaceImpl<A> extends AbstractInterfaceImpl<A> {
    @Override
    public String test(String name) {
        return "Name is " + name;
    }

    @Override
    public String test(int age) {
        return "Age is " + age;
    }


    @Override
    public String test() {
        return "noargs";
    }

    @Override
    public void testVoid(String name) {
        assert name.equals("changed");
    }

    @Override
    public void testVoid(String name, int age) {
        assert name.equals("changed");
        assert age == 10;
    }

    @Override
    public boolean testBoolean(String name) {
        return name.equals("changed");
    }

    @Override
    public boolean testBoolean(String name, int age) {
        assert age == 10;
        return name.equals("changed");
    }

    @Override
    public int testInt(String name) {
        return name.equals("changed") ? 1 : 0;
    }

    @Override
    public int testInt(String name, int age) {
        assert name.equals("test");
        return age;
    }

    @Override
    public long testLong(String name) {
        return name.equals("changed") ? 1 : 0;
    }

    @Override
    public long testLong(String name, int age) {
        assert name.equals("test");
        return age;
    }

    @Override
    public short testShort(String name) {
        return (short) (name.equals("changed") ? 1 : 0);
    }

    @Override
    public short testShort(String name, int age) {
        assert name.equals("test");
        return (short) age;
    }

    @Override
    public byte testByte(String name) {
        return (byte) (name.equals("changed") ? 1 : 0);
    }

    @Override
    public byte testByte(String name, int age) {
        assert name.equals("test");
        return (byte) age;
    }

    @Override
    public double testDouble(String name) {
        return name.equals("changed") ? 1 : 0;
    }

    @Override
    public double testDouble(String name, int age) {
        assert name.equals("test");
        return age;
    }

    @Override
    public float testFloat(String name) {
        return name.equals("changed") ? 1 : 0;
    }

    @Override
    public float testFloat(String name, int age) {
        assert name.equals("test");
        return age;
    }

    @Override
    public char testChar(String name) {
        return (char) (name.equals("changed") ? 1 : 0);
    }

    @Override
    public char testChar(String name, int age) {
        assert name.equals("test");
        return (char) age;
    }

    @Override
    public byte[] testByteArray(String name, byte[] data) {
        assert name.equals("changed");
        return data;
    }

    @Override
    public <T extends CharSequence> T testGenericsWithExtends(T name, int age) {
        return (T) ("Name is " + name);
    }

    @Override
    public <T> List<? super String> testListWithWildCardSuper(T name, List<? super String> p2) {
        return Collections.singletonList(name.toString());
    }

    @Override
    public <T> List<? extends String> testListWithWildCardExtends(T name, List<? extends String> p2) {
        return Collections.singletonList(name.toString());
    }

    @Override
    public A testGenericsFromType(A name, int age) {
        return (A) ("Name is " + name);
    }
}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/aop/itfce/InterfaceImpl.kt
package io.micronaut.kotlin.processing.aop.itfce

import io.micronaut.kotlin.processing.aop.simple.CovariantClass
import jakarta.inject.Singleton

@Singleton
open class InterfaceImpl<A> : AbstractInterfaceImpl<A>() {

    override fun test(name: String): String {
        return "Name is $name"
    }

    override fun test(age: Int): String {
        return "Age is $age"
    }

    override fun test(): String {
        return "noargs"
    }

    override fun testVoid(name: String) {
        assert(name == "changed")
    }

    override fun testVoid(name: String, age: Int) {
        assert(name == "changed")
        assert(age == 10)
    }

    override fun testBoolean(name: String): Boolean {
        return name == "changed"
    }

    override fun testBoolean(name: String, age: Int): Boolean {
        assert(age == 10)
        return name == "changed"
    }

    override fun testInt(name: String): Int {
        return if (name == "changed") 1 else 0
    }

    override fun testInt(name: String, age: Int): Int {
        assert(name == "test")
        return age
    }

    override fun testLong(name: String): Long {
        return if (name == "changed") 1 else 0
    }

    override fun testLong(name: String, age: Int): Long {
        assert(name == "test")
        return age.toLong()
    }

    override fun testShort(name: String): Short {
        return (if (name == "changed") 1 else 0).toShort()
    }

    override fun testShort(name: String, age: Int): Short {
        assert(name == "test")
        return age.toShort()
    }

    override fun testByte(name: String): Byte {
        return (if (name == "changed") 1 else 0).toByte()
    }

    override fun testByte(name: String, age: Int): Byte {
        assert(name == "test")
        return age.toByte()
    }

    override fun testDouble(name: String): Double {
        return if (name == "changed") 1.0 else 0.0
    }

    override fun testDouble(name: String, age: Int): Double {
        assert(name == "test")
        return age.toDouble()
    }

    override fun testFloat(name: String): Float {
        return if (name == "changed") 1F else 0F
    }

    override fun testFloat(name: String, age: Int): Float {
        assert(name == "test")
        return age.toFloat()
    }

    override fun testChar(name: String): Char {
        return (if (name == "changed") 1 else 0).toChar()
    }

    override fun testChar(name: String, age: Int): Char {
        assert(name == "test")
        return age.toChar()
    }

    override fun testByteArray(name: String, data: ByteArray): ByteArray {
        assert(name == "changed")
        return data
    }

    override fun <T : CharSequence> testGenericsWithExtends(name: T, age: Int): T {
        return "Name is $name" as T
    }

    override fun <T> testListWithWildCardIn(name: T, p2: CovariantClass<in String>): CovariantClass<in String> {
        return CovariantClass(name.toString())
    }

    override fun <T> testListWithWildCardOut(name: T, p2: CovariantClass<out String>): CovariantClass<out String> {
        return CovariantClass(name.toString())
    }

    override fun testGenericsFromType(name: A, age: Int): A {
        return "Name is $name" as A
    }
}
